tools/checklocks/checklocks.go

```diff
@@ -138,8 +152,22 @@ func run(pass *analysis.Pass) (interface{}, error) {
 			continue
 		}
 
+		if Benchmark {
+			start = time.Now()
+		}
+
 		// Check the basic blocks in the function.
 		pc.checkFunction(nil, fn, &lff, nil, false /* force */)
+
+		if Benchmark {
+			end := time.Since(start)
+			pc.pass.ImportPackageFact(pc.pass.Pkg, &pgf)
```

Comment:
This is potentially very expensive if it gets fully serialized and deserialized. It is a lot cheaper to just carry pgf around.